traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -897,3 +955,483 @@ last_updated`
+		}
+		log.Debugln("IMS MISS")
+	} else {
+		log.Debugln("Non IMS request")
+	}
+	baseSelect := SelectQuery()
+	if _, ok := params["topology"]; ok {
+		baseSelect += `
+		LEFT JOIN topology_cachegroup ON cachegroup.name = topology_cachegroup.cachegroup
+		`
+	}
+	// If the type cannot be converted to an int, return 400
+	if cgType, ok := params["type"]; ok {
+		_, err := strconv.Atoi(cgType)
+		if err != nil {
+			return nil, time.Time{}, http.StatusBadRequest, nil, fmt.Errorf("cachegroup read: converting cachegroup type to integer " + err.Error())
+		}
+	}
+
+	query := baseSelect + where + orderBy + pagination
+	rows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		return nil, time.Time{}, http.StatusInternalServerError, nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var cg TOCacheGroupV5
+		lms := make([]tc.LocalizationMethod, 0)
+		cgfs := make([]string, 0)
+		if err = rows.Scan(
+			&cg.ID,
+			&cg.Name,
+			&cg.ShortName,
+			&cg.Latitude,
+			&cg.Longitude,
+			pq.Array(&lms),
+			&cg.ParentCachegroupID,
+			&cg.ParentName,
+			&cg.SecondaryParentCachegroupID,
+			&cg.SecondaryParentName,
+			&cg.Type,
+			&cg.TypeID,
+			&cg.LastUpdated,
+			pq.Array(&cgfs),
+			&cg.FallbackToClosest,
+		); err != nil {
+			return nil, time.Time{}, http.StatusInternalServerError, nil, err
+		}
+		cg.LocalizationMethods = &lms
+		cg.Fallbacks = &cgfs
+		cgList = append(cgList, cg)
+	}
+
+	return cgList, maxTime, http.StatusOK, nil, nil
+}
+
+// CreateCacheGroup [Version : V5] function creates the cache group with the passed name.
+func CreateCacheGroup(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	tx := inf.Tx.Tx
+
+	cg, readValErr := readAndValidateJsonStruct(r)
+	if readValErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, readValErr, nil)
+		return
+	}
+
+	// check if cache group already exists
+	var exists bool
+	err := tx.QueryRow(`SELECT EXISTS(SELECT * from cachegroup where name = $1)`, cg.Name).Scan(&exists)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("error: %w, when checking if cache group with name %s exists", err, cg.Name))
+		return
+	}
+	if exists {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("cache group name '%s' already exists.", cg.Name), nil)
+		return
+	}
+
+	// create cache group
+	query := InsertQuery()
+
+	err = tx.QueryRow(
+		query,
+		cg.Name,
+		cg.ShortName,
+		cg.TypeID,
+		cg.ParentCachegroupID,
+		cg.SecondaryParentCachegroupID,
+		cg.FallbackToClosest,
+	).Scan(
+		&cg.ID,
+		&cg.Type,
+		&cg.ParentName,
+		&cg.SecondaryParentName,
+	)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, fmt.Errorf("error: %w in creating cache group with name: %s", err, cg.Name), nil)
```

Comment:
Since `cg.Name` is a pointer this will print out the address instead of the cachegroup name, since the cachegroup has already been validated you just need to dereference e.g. `*cg.Name`

Reply:
Fixed